### PR TITLE
Add data decoder/master copy endpoint calls

### PR DESCRIPTION
**Pull request by iamacook**
In removing the transaction service calls here: https://github.com/gnosis/safe-react/issues/3114 it was possible to move the `data-decoder` and `master-copies` endpoint interaction to the SDK.

In this issue, the types were also cleaned up.

### CODE REVIEW

File: src/types/master-copies.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+export enum MasterCopyDeployer {
+  GNOSIS = 'Gnosis',
+  CIRCLES = 'Circles',
+}
+
+export type MasterCopyReponse = {
+  address: string
+  version: string
+  deployer: MasterCopyDeployer
```
Comment (DaniSomoza) [inline src/types/master-copies.ts:9]: I only see this 2 keys `address` and `version` in the response. Keys `deployer` and `deployerRepoUrl` exists? if they exists we can set them as optional?

Comment (iamacook) [reply src/types/master-copies.ts]: Good catch! I copied these over from our client-side types and they are indeed not present in the response.

**Comment by DaniSomoza** [comment]
What is your opinion about to add e2e test to those new endpoint?

`get-master-copy.test.ts`

```
import { getMasterCopies } from '../src'
import config from './config'

describe('get mastercopy tests', () => {
  it('should get master copy response', async () => {
    const masterCopies = await getMasterCopies(config.baseUrl, '4')

    expect(Array.isArray(masterCopies)).toBe(true)

    // version 1.1.1 should be present
    const lastVersionRinkeby = masterCopies.find((mastercopy) => mastercopy.version === '1.1.1')
    expect(lastVersionRinkeby).toBeDefined()
    expect(lastVersionRinkeby.address).toBe('0x34CfAC646f301356fAa8B21e94227e3583Fe3F5F')
  })
})
```

**Review by DaniSomoza — APPROVED** [review]
Approved! :D

**Pull request merged**